{
    "links": {
        "decline": {
            "href": "https://api.bitbucket.org/2.0/repositories/yt_analysis/yt/pullrequests/1528/decline"
        },
        "approve": {
            "href": "https://api.bitbucket.org/2.0/repositories/yt_analysis/yt/pullrequests/1528/approve"
        },
        "html": {
            "href": "https://bitbucket.org/yt_analysis/yt/pull-requests/1528"
        },
        "activity": {
            "href": "https://api.bitbucket.org/2.0/repositories/yt_analysis/yt/pullrequests/1528/activity"
        },
        "self": {
            "href": "https://api.bitbucket.org/2.0/repositories/yt_analysis/yt/pullrequests/1528"
        },
        "commits": {
            "href": "https://api.bitbucket.org/2.0/repositories/yt_analysis/yt/pullrequests/1528/commits"
        },
        "merge": {
            "href": "https://api.bitbucket.org/2.0/repositories/yt_analysis/yt/pullrequests/1528/merge"
        },
        "statuses": {
            "href": "https://api.bitbucket.org/2.0/repositories/yt_analysis/yt/pullrequests/1528/statuses"
        },
        "comments": {
            "href": "https://api.bitbucket.org/2.0/repositories/yt_analysis/yt/pullrequests/1528/comments"
        },
        "diff": {
            "href": "https://api.bitbucket.org/2.0/repositories/yt_analysis/yt/pullrequests/1528/diff"
        }
    },
    "task_count": 0,
    "source": {
        "repository": {
            "full_name": "chummels/yt",
            "uuid": "{cc72b7fe-3d37-4cd2-a367-f2de113a2e82}",
            "type": "repository",
            "links": {
                "self": {
                    "href": "https://api.bitbucket.org/2.0/repositories/chummels/yt"
                },
                "html": {
                    "href": "https://bitbucket.org/chummels/yt"
                },
                "avatar": {
                    "href": "https://bitbucket.org/chummels/yt/avatar/32/"
                }
            },
            "name": "yt"
        },
        "commit": {
            "hash": "e0eb6bfb0c80",
            "links": {
                "self": {
                    "href": "https://api.bitbucket.org/2.0/repositories/chummels/yt/commit/e0eb6bfb0c80"
                }
            }
        },
        "branch": {
            "name": "yt"
        }
    },
    "closed_by": {
        "username": "ngoldbaum",
        "type": "user",
        "links": {
            "self": {
                "href": "https://api.bitbucket.org/2.0/users/ngoldbaum"
            },
            "html": {
                "href": "https://bitbucket.org/ngoldbaum/"
            },
            "avatar": {
                "href": "https://bitbucket.org/account/ngoldbaum/avatar/32/"
            }
        },
        "display_name": "Nathan Goldbaum",
        "uuid": "{5dcd72da-6420-4a6d-9571-e75dffcac587}"
    },
    "title": "[bugfix] Providing better error messages to PlotWindow callbacks",
    "id": 1528,
    "comment_count": 4,
    "created_on": "2015-03-30T16:34:56.875103+00:00",
    "merge_commit": {
        "hash": "55e26d7731c6",
        "links": {
            "self": {
                "href": "https://api.bitbucket.org/2.0/repositories/yt_analysis/yt/commit/55e26d7731c6"
            }
        }
    },
    "comments": [
        {
            "content": {
                "markup": "markdown",
                "raw": "All tests pass. Good job!",
                "html": "<p>All tests pass. Good job!</p>"
            },
            "user": {
                "username": "yt-fido",
                "type": "user",
                "links": {
                    "self": {
                        "href": "https://api.bitbucket.org/2.0/users/yt-fido"
                    },
                    "html": {
                        "href": "https://bitbucket.org/yt-fido/"
                    },
                    "avatar": {
                        "href": "https://bitbucket.org/account/yt-fido/avatar/32/"
                    }
                },
                "display_name": "yt-fido",
                "uuid": "{da2aa332-3f05-45c9-ae53-ff5ba34b35ce}"
            },
            "links": {
                "self": {
                    "href": "https://api.bitbucket.org/2.0/repositories/yt_analysis/yt/pullrequests/1528/comments/5963862"
                },
                "html": {
                    "href": "https://bitbucket.org/yt_analysis/yt/pull-requests/1528/_/diff#comment-5963862"
                }
            },
            "created_on": "2015-03-30T17:18:40.068558+00:00",
            "updated_on": "2015-03-30T17:18:40.282467+00:00",
            "pullrequest": {
                "type": "pullrequest",
                "links": {
                    "self": {
                        "href": "https://api.bitbucket.org/2.0/repositories/yt_analysis/yt/pullrequests/1528"
                    },
                    "html": {
                        "href": "https://bitbucket.org/yt_analysis/yt/pull-requests/1528"
                    }
                },
                "title": "[bugfix] Providing better error messages to PlotWindow callbacks",
                "id": 1528
            },
            "type": "pullrequest_comment",
            "id": 5963862
        },
        {
            "content": {
                "markup": "markdown",
                "raw": "Thanks for looking at this Cameron!  I made a couple minor modifications on top of this, see [a0add41](https://bitbucket.org/ngoldbaum/yt/commits/a0add41bed4dcd1f2c7ab2cea4142a83f4ca8c6d?at=plot_callback_error_message) in my repo.\n\nBasically, I think we want to re-raise with the original traceback to make debugging a bit easier.  I also think we should be referring to the callback with the `annotate_foo` name that the user knows about rather than the internal `FooCallback` name.",
                "html": "<p>Thanks for looking at this Cameron!  I made a couple minor modifications on top of this, see <a href=\"https://bitbucket.org/ngoldbaum/yt/commits/a0add41bed4dcd1f2c7ab2cea4142a83f4ca8c6d?at=plot_callback_error_message\">a0add41</a> in my repo.</p>\n<p>Basically, I think we want to re-raise with the original traceback to make debugging a bit easier.  I also think we should be referring to the callback with the <code>annotate_foo</code> name that the user knows about rather than the internal <code>FooCallback</code> name.</p>"
            },
            "user": {
                "username": "ngoldbaum",
                "type": "user",
                "links": {
                    "self": {
                        "href": "https://api.bitbucket.org/2.0/users/ngoldbaum"
                    },
                    "html": {
                        "href": "https://bitbucket.org/ngoldbaum/"
                    },
                    "avatar": {
                        "href": "https://bitbucket.org/account/ngoldbaum/avatar/32/"
                    }
                },
                "display_name": "Nathan Goldbaum",
                "uuid": "{5dcd72da-6420-4a6d-9571-e75dffcac587}"
            },
            "links": {
                "self": {
                    "href": "https://api.bitbucket.org/2.0/repositories/yt_analysis/yt/pullrequests/1528/comments/5964294"
                },
                "html": {
                    "href": "https://bitbucket.org/yt_analysis/yt/pull-requests/1528/_/diff#comment-5964294"
                }
            },
            "created_on": "2015-03-30T17:35:48.400961+00:00",
            "updated_on": "2015-03-30T17:36:02.442675+00:00",
            "pullrequest": {
                "type": "pullrequest",
                "links": {
                    "self": {
                        "href": "https://api.bitbucket.org/2.0/repositories/yt_analysis/yt/pullrequests/1528"
                    },
                    "html": {
                        "href": "https://bitbucket.org/yt_analysis/yt/pull-requests/1528"
                    }
                },
                "title": "[bugfix] Providing better error messages to PlotWindow callbacks",
                "id": 1528
            },
            "type": "pullrequest_comment",
            "id": 5964294
        },
        {
            "content": {
                "markup": "markdown",
                "raw": "Sounds good to me.  I pulled your changes in, and fixed a small typo.  Looks good.  Thanks for the help!",
                "html": "<p>Sounds good to me.  I pulled your changes in, and fixed a small typo.  Looks good.  Thanks for the help!</p>"
            },
            "user": {
                "username": "chummels",
                "type": "user",
                "links": {
                    "self": {
                        "href": "https://api.bitbucket.org/2.0/users/chummels"
                    },
                    "html": {
                        "href": "https://bitbucket.org/chummels/"
                    },
                    "avatar": {
                        "href": "https://bitbucket.org/account/chummels/avatar/32/"
                    }
                },
                "display_name": "chummels",
                "uuid": "{de871bb8-32e9-450a-b8c9-2b49a1334f16}"
            },
            "parent": {
                "links": {
                    "self": {
                        "href": "https://api.bitbucket.org/2.0/repositories/yt_analysis/yt/pullrequests/1528/comments/5964294"
                    },
                    "html": {
                        "href": "https://bitbucket.org/yt_analysis/yt/pull-requests/1528/_/diff#comment-5964294"
                    }
                },
                "id": 5964294
            },
            "created_on": "2015-03-30T20:11:19.825584+00:00",
            "links": {
                "self": {
                    "href": "https://api.bitbucket.org/2.0/repositories/yt_analysis/yt/pullrequests/1528/comments/5967875"
                },
                "html": {
                    "href": "https://bitbucket.org/yt_analysis/yt/pull-requests/1528/_/diff#comment-5967875"
                }
            },
            "pullrequest": {
                "type": "pullrequest",
                "links": {
                    "self": {
                        "href": "https://api.bitbucket.org/2.0/repositories/yt_analysis/yt/pullrequests/1528"
                    },
                    "html": {
                        "href": "https://bitbucket.org/yt_analysis/yt/pull-requests/1528"
                    }
                },
                "title": "[bugfix] Providing better error messages to PlotWindow callbacks",
                "id": 1528
            },
            "type": "pullrequest_comment",
            "id": 5967875,
            "updated_on": "2015-03-30T20:11:28.556925+00:00"
        },
        {
            "content": {
                "markup": "markdown",
                "raw": "All tests pass. Good job!",
                "html": "<p>All tests pass. Good job!</p>"
            },
            "user": {
                "username": "yt-fido",
                "type": "user",
                "links": {
                    "self": {
                        "href": "https://api.bitbucket.org/2.0/users/yt-fido"
                    },
                    "html": {
                        "href": "https://bitbucket.org/yt-fido/"
                    },
                    "avatar": {
                        "href": "https://bitbucket.org/account/yt-fido/avatar/32/"
                    }
                },
                "display_name": "yt-fido",
                "uuid": "{da2aa332-3f05-45c9-ae53-ff5ba34b35ce}"
            },
            "links": {
                "self": {
                    "href": "https://api.bitbucket.org/2.0/repositories/yt_analysis/yt/pullrequests/1528/comments/5969556"
                },
                "html": {
                    "href": "https://bitbucket.org/yt_analysis/yt/pull-requests/1528/_/diff#comment-5969556"
                }
            },
            "created_on": "2015-03-30T21:30:43.076498+00:00",
            "updated_on": "2015-03-30T21:30:43.125025+00:00",
            "pullrequest": {
                "type": "pullrequest",
                "links": {
                    "self": {
                        "href": "https://api.bitbucket.org/2.0/repositories/yt_analysis/yt/pullrequests/1528"
                    },
                    "html": {
                        "href": "https://bitbucket.org/yt_analysis/yt/pull-requests/1528"
                    }
                },
                "title": "[bugfix] Providing better error messages to PlotWindow callbacks",
                "id": 1528
            },
            "type": "pullrequest_comment",
            "id": 5969556
        }
    ],
    "reason": "",
    "type": "pullrequest",
    "close_source_branch": false,
    "state": "MERGED",
    "description": "This is in response to some of the issues that came up on the mailing list last week.  Previously, callbacks gave somewhat cryptic error messages, as it didn't identify *which* callback was failing in the traceback.  By adding a `YTPlotCallbackException`, this is now resolved.\r\n\r\nExample when specifying 2 dimensional positions for `annotate_marker()` callback using default `data` coords:\r\n\r\nhttp://paste.yt-project.org/show/5500/",
    "author": {
        "username": "chummels",
        "type": "user",
        "links": {
            "self": {
                "href": "https://api.bitbucket.org/2.0/users/chummels"
            },
            "html": {
                "href": "https://bitbucket.org/chummels/"
            },
            "avatar": {
                "href": "https://bitbucket.org/account/chummels/avatar/32/"
            }
        },
        "display_name": "chummels",
        "uuid": "{de871bb8-32e9-450a-b8c9-2b49a1334f16}"
    },
    "updated_on": "2015-03-30T21:31:34.604770+00:00",
    "destination": {
        "repository": {
            "full_name": "yt_analysis/yt",
            "uuid": "{b73cb37c-40f2-493a-aaa9-41bc8d8974cc}",
            "type": "repository",
            "links": {
                "self": {
                    "href": "https://api.bitbucket.org/2.0/repositories/yt_analysis/yt"
                },
                "html": {
                    "href": "https://bitbucket.org/yt_analysis/yt"
                },
                "avatar": {
                    "href": "https://bitbucket.org/yt_analysis/yt/avatar/32/"
                }
            },
            "name": "yt"
        },
        "commit": {
            "hash": "db1ad1f832ed",
            "links": {
                "self": {
                    "href": "https://api.bitbucket.org/2.0/repositories/yt_analysis/yt/commit/db1ad1f832ed"
                }
            }
        },
        "branch": {
            "name": "yt"
        }
    }
}